### PR TITLE
Device Type

### PR DESCRIPTION
**Pull request by jonathanaustin**
Addresses #803.

Provide access to the backing `HttpServletRequest` in `ServletRequest`.

A new helper method in `ServletUtil` to determine the user’s device.

### CODE REVIEW

**Comment by jonathanaustin** [comment]
PTAL @marksreeves @ricksbrown

File: wcomponents-core/src/test/java/com/github/bordertech/wcomponents/servlet/ServletUtilDeviceType_Test.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import org.junit.runners.Parameterized;
+
+/**
+ * Parameterized test for device types in {@link ServletUtil}.
+ *
+ * @author Jonathan Austin
+ * @since 1.2.5
+ */
+@RunWith(Parameterized.class)
+public class ServletUtilDeviceType_Test {
```
Comment (ricksbrown) [inline wcomponents-core/src/test/java/com/github/bordertech/wcomponents/servlet/ServletUtilDeviceType_Test.java]: I like the tests. 

Do you think we should add UC user agent string to the list?

Also some of those user agent strings look like templates, for example `<Build Tag etc.>` do you think we should actually file them out or turn them into sprintf strings?

**Comment by ricksbrown** [comment]
LGTM

File: wcomponents-core/src/test/java/com/github/bordertech/wcomponents/servlet/ServletUtilDeviceType_Test.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	private static final String MOBILE_SAFARI = "Mozilla/5.0 (iPad; CPU OS 9_3_3 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13G34 Safari/601.1";
+
+	private static final String CHROME_WIN_10 = "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36";
+
+	private static final String CHROME_MOBILE = "Mozilla/5.0 (Linux; <Android Version>; <Build Tag etc.>) AppleWebKit/<WebKit Rev> (KHTML, like Gecko) Chrome/<Chrome Rev> Mobile Safari/<WebKit Rev>";
```
Comment (marksreeves) [inline wcomponents-core/src/test/java/com/github/bordertech/wcomponents/servlet/ServletUtilDeviceType_Test.java]: The parts of the user agent strings in angle bracket are, indeed, templates. They are not _really_ important for the purposes of these tests. If you want to replace them with some actual common user agent strings see http://www.useragentstring.com/pages/useragentstring.php

**Comment by marksreeves** [comment]
LGTM

**Comment by jonathanaustin** [comment]
We can definitely add more User Agent Strings as we go. I will merge the change.

**Pull request merged**